classes/form/cms_types_form.php

```diff
@@ -52,6 +54,27 @@ protected function definition() {
         $mform->addElement('editor', 'description', get_string('description'));
         $mform->setType('description', PARAM_RAW);
 
+        $mform->addElement(
+            'textarea',
+            'mustache',
+            get_string('mustache', 'cms'),
+            ['rows' => 10]
+        );
+        $mform->setType('mustache', PARAM_RAW);
+
+        // Generate the help text for mustache template.
+        $cmstype = new cms_types();
+        $cms = $cmstype->get_sample_cms();
+        $renderer = new renderer($cms);
+        $data = $renderer->get_data();
+        $syntaxlink = \html_writer::link(
+            new \moodle_url('https://moodledev.io/docs/guides/templates'),
+            get_string('mustache_template', 'cms')
+        );
+        $helptext = get_string('mustache_help', 'cms', $syntaxlink);
+        $helptext .= \html_writer::tag('pre', json_encode($data, JSON_PRETTY_PRINT));
+        $mform->addElement('static', 'mustache_help', '', $helptext);
```

Comment:
can you please make a 'save and continue' and a 'save and exit' buttons

Reply:
done